src/file.cpp

```diff
@@ -1111,6 +1149,30 @@ namespace libtorrent
 		remove(f, ec);
 	}
 
+	void remove_empty(std::string const& f, error_code& ec)
+	{
+		ec.clear();
+
+		file_status s;
+		stat_file(f, &s, ec);
+		if (ec) return;
+
+		if (s.mode & file_status::directory)
+		{
+			for (directory i(f, ec); !i.done(); i.next(ec))
+			{
+				if (ec) return;
+
+				std::string p = i.file();
+				if (p == "." || p == "..") continue;
+				remove_empty(combine_path(f, p), ec);
+				ec.clear();
+			}
+
+			remove(f, ec);
```

Comment:
I just see an empty folder as something that can be safely removed in any case. Although, of course, we cannot be sure of this.